### PR TITLE
CLI config file

### LINKED ISSUE #1175: CLI configuration file
The CLI configuration file should follow [FSH](http://www.pathname.com/fhs/pub/fhs-2.3.html#HOMEUSERHOMEDIRECTORIES) convention as mentioned in issue #1153 and should therefore be a dot directory called `.amp`.
This directory will be looked up using this sequence:
 - current folder(`./.amp`),
 - user home directory (`$HOME/.config/amp`).

The content of the `.amp` directory is:
 - `amp`: the configuration file for the CLI.
 - `$serverAddress.credentials`: The credentials used for the given serverAddress.

Note that credentials should only be stored in user's home directory.

### PR DESCRIPTION
**Pull request by neha-viswanathan**
Resolves #1175 

The smoke tests are located in `platform/tests/config` 

You should see a `localhost.credentials` file in $HOME/.config/amp directory
If no server address is specified on the command, the server settings specified in the CLI config file is read. 
Note that the CLI config file can be found in one of the following locations - 
- current working directory (`./.amp`)
- user's HOME directory  #(`$HOME/.config/amp`)

If no config file is found or no server address is specified on the command, the default server address is used.

### CODE REVIEW

**Review by bquenin — DISMISSED** [review]
LGTM

File: platform/tests/cli/configuration [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+mkdir -p $PWD/.amp
+echo "Server: present" > $PWD/.amp/amp.yml
+
+# for config file in home directory
+mkdir -p $HOME/.config/amp
+echo "Server: home" > $HOME/.config/amp/amp.yml
+
+# no config file; server address passed as command line argument
+amp $addr version | grep -Eq "Server:[[:space:]]+command-line"
+if [[ $? -ne 0 ]]; then
```
Comment (ndegory) [inline platform/tests/cli/configuration]: if you use `set -e` this will never be executed.

File: platform/tests/config/config_test.sh [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+  # for config file in home directory
+  mkdir -p $HOME/.config/amp
+  echo "Server: home" > $HOME/.config/amp/amp.yml
+
+  # server address passed as command line argument
+  amp $addr version | grep -Eq "Server:[[:space:]]+command-line"
+
+  rm -R $PWD/.amp
+  rm -R $HOME/.config/amp
```
Comment (subfuzion) [inline platform/tests/config/config_test.sh]: This won't work. The test will pass as long as the last command passes. If you want to clean up at the end of your test, just make sure to preserve the result of the previous grep (`local ec=$?`) and then return it at the end of the function (`return $ec`).

**Review by bquenin — APPROVED** [review]
LGTM

**Pull request merged**